src/ClientNet462/Program.cs

```diff
@@ -0,0 +1,148 @@
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Client
+{
+    internal class Program
+    {
+        static async Task Main(string[] args)
+        {
+            try
+            {
+                using (var cts = new CancellationTokenSource())
+                {
+                    ConsoleCancelEventHandler handler = (o, e) =>
+                    {
+                        Console.WriteLine("Cancelling...");
+                        cts.CancelAfter(0);
+                        e.Cancel = true;
```

Comment:
`cts.Cancel();` executes synchronously, throwing a `TaskCancellationException` on the other thread, which hits the finally block to disassociate the `CancelKeyPress` handler, which results in a deadlock!!!!  ugh!  Solved with `CancelAfter(0);`.  Also, without `e.Cancel = true;`, the process immediately terminates, which appears as a hanged application when debugging.

Reply:
WOW!

Reply:
Yeah, sometimes multithreading really comes back to bite you! - lol

Reply:
Well,  `Console.CancelKeyPress -= handler;` actually is not needed for such run-and-terminate console app. Does deadlock occur without it?

Reply:
no, but if a user was to press Ctrl-C afterwards it would throw an ObjectDisposedException when accessing the CancellationTokenSource.  I figured as a matter of 'good form' it should show proper removal of the handler prior to disposing of the cts.  Of course I could have either (a) not disposed of the cts at all, or (b) put the other console code within the using block.

Reply:
I just figured that it's always best to have samples with proper tear-down code.

Reply:
Now realizing that `Cancel` is synchronous, it probably should properly be `CancelAfter` anyway in this sample, in case of any other locks put in place by `Console`.

Reply:
What mean "Cancel is synchronous"?

Reply:
`Cancel()` is a blocking operation that executes all registered cancellation notification delegates.  The docs say it can throw an `AggregateException` if any delegate throws an exception, so this is known behavior.  Apparently in this case the registered cancellation delegate within the `HttpClient` is also executing a blocking operation when throwing exceptions on pending `Task`s.  The `finally` clause then creates the deadlock attempting to edit an event handler while `Console` is executing the handlers (probably placing a lock on the list while iterating through it).

See: https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.cancel?view=net-6.0#System_Threading_CancellationTokenSource_Cancel

Reply:
> registered cancellation notification delegates

These are those using `cts.Token.Register(() => throw new Exception("aaa"));`. Example has no such registrations (I don't see) so nothing bad can happen. I have a few CLI tools (Core, not NET) that use such approach with `CancellationTokenSource` and `ConsoleKeyPress` handler. They work without problems, the cancellation works, there are no dedlocks. I do not know... maybe it's NET differs from Core. But for the purposes of the example, it is not important.